### PR TITLE
FAB Snippet for API Docs

### PR DESCRIPTION
**Pull request by Piinks**
## Description

This PR adds 2 code snippets for the FloatingActionButton class in the API docs. The first is a simple case, the second illustrates the use of the extended FAB constructor.

**For your consideration dear reviewer...**
I used the stateless widget template, and left the onPressed properties empty in these samples to focus on a simple example of constructing the FAB widget.
If changing this template to a stateful widget and implementing the popular counter example (or something similar) would be better to further exemplify this widget, share your thoughts below and I can modify it!  

## Related Issues

Ref: #21136

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/floating_action_button.dart [hunk 1]
```diff
@@ -53,6 +53,58 @@ class _DefaultHeroTag {
 /// disabled. Consider changing the [backgroundColor] if disabling the floating
 /// action button.
 ///
+/// {@tool snippet --template=stateless_widget_material}
+/// This example shows how to make a simple [FloatingActionButton] in a
+/// [Scaffold], with a  pink [backgroundColor] and a thumbs up [Icon].
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/floating_action_button.dart]:
```suggestion
/// [Scaffold], with a pink [backgroundColor] and a thumbs up [Icon].
```

File: packages/flutter/lib/src/material/floating_action_button.dart [hunk 2]
```diff
@@ -53,6 +53,58 @@ class _DefaultHeroTag {
...
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     appBar: AppBar(
+///       title: Text('Floating Action Button Sample'),
+///     ),
+///     body: Center(
+///         child: Text('Press the button below!')
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/floating_action_button.dart]:
```suggestion
///       child: Text('Press the button below!')
```

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

(modulo whitespace nits)

**Comment by Piinks** [comment]
Interface isn't updating analyze pre-submit check result. All green on cirrus, checked by @dnfield.

**Pull request merged**